data/src/main/java/io/homeassistant/companion/android/data/integration/IntegrationRepositoryImpl.kt

```diff
@@ -126,6 +126,29 @@ class IntegrationRepositoryImpl @Inject constructor(
         throw IntegrationException()
     }
 
+    override suspend fun callService(domain: String, service: String, serviceData: HashMap<String, String>) {
+        var wasSuccess = false
+
+        val serviceCallRequest = ServiceCallRequest(domain, service, serviceData)
+
+        for (it in urlRepository.getApiUrls()) {
+            try {
+                wasSuccess =
+                    integrationService.callService(
+                        it.toHttpUrlOrNull()!!,
+                        IntegrationRequest("call_service", serviceCallRequest)
+                    ).isSuccessful
+            } catch (e: Exception) {
+                // Ignore failure until we are out of URLS to try!
+            }
+            // if we had a successful call we can return
+            if (wasSuccess)
+                return
+        }
```

Comment:
You don't need to try catch the casting, just first map the URLs using a `mapNotNull` which maps and filters `null`s
```kotlin
urlRepository.getApiUrls().mapNotNull { it.toHttpUrlOrNull() }.forEach {
// ...
```

Reply:
While I'm not going to disagree that this might be more elegant solution, the formatting for this method was taken from `updateLocation(...)`, `updateRegistration(...)`, and `getZones()`.  If I change it here, it should probably be changed in those as well.